common/include/common/binary/base_serializer.hpp

```diff
@@ -0,0 +1,83 @@
+#ifndef RCBE_ENGINE_BASE_SERIALIZER_HPP
+#define RCBE_ENGINE_BASE_SERIALIZER_HPP
+
+#include <cstddef>
+#include <cstdint>
+#include <stdexcept>
+
+namespace rcbe::binary {
+template <typename Buffer, typename ActualType>
+void to_binary(Buffer &c, const ActualType &a) {
+    throw std::runtime_error("Please provide to_binary implementation for the type " + std::string(typeid(ActualType).name()) + "!");
```

Comment:
This'll be unreadable, since it would be mangled. Either demangle it (in crossplatform fashion), or discard this part of the message.